### PR TITLE
EVP_Cipher: correct parameter checking

### PR DESCRIPTION
**Pull request by julek-wolfssl**
- EVP_Cipher(ctx, NULL, NULL, 0) is a valid call for all algorithms. For none-AEAD it results in a no-op.
- Added libssh2 test

Fixes ZD17303

### CODE REVIEW

File: wolfcrypt/src/evp.c [hunk 1]
```diff
@@ -8141,9 +8145,15 @@ void wolfSSL_EVP_init(void)
         #ifdef WOLFSSL_SM4_CCM
             && ctx->cipherType != SM4_CCM_TYPE
         #endif
-            ))) {
-            WOLFSSL_MSG("Bad argument.");
-            return WOLFSSL_FATAL_ERROR;
+            ) {
+            /* Not an AEAD cipher */
+            /* No-op for none AEAD ciphers */
```
Comment (SparkiDev) [inline wolfcrypt/src/evp.c:8150]: 'none AEAD' -> 'non-AEAD'

File: wolfcrypt/src/evp.c [hunk 2]
```diff
@@ -8118,8 +8118,12 @@ void wolfSSL_EVP_init(void)
...
 
-        if (ctx == NULL || ((src == NULL || dst == NULL) &&
-            (TRUE
+        if (ctx == NULL) {
+            WOLFSSL_MSG("Bad argument.");
+            return WOLFSSL_FATAL_ERROR;
+        }
+
+        if (TRUE
         #ifdef HAVE_AESGCM
```
Comment (SparkiDev) [inline wolfcrypt/src/evp.c:8127]: Make the check as to whether the ctx is an AEAD algorithm a new function.
Will clean up this code and potentially be useable elsewhere.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/evp.c (hunk 1 above)
Comment (julek-wolfssl) [reply wolfcrypt/src/evp.c]: Fixed.

File: wolfcrypt/src/evp.c (hunk 2 above)
Comment (julek-wolfssl) [reply wolfcrypt/src/evp.c]: Refactored.

Comment (SparkiDev) [reply wolfcrypt/src/evp.c]: Thanks!

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**